Tpetra:  CrsMatrix fails to compile when LocalOrdinal != int

## Bug Report
@trilinos/tpetra

### Description
Hi,

I've encountered a couple compilation errors when trying to instantiate and use `Tpetra::CrsMatrix<>` template with `LocalOrdinal = std::ptrdiff_t` (which on my system resolves to `long int`). The errors reproduced on 12.18.1 release, but relevant code is still there in master branch. The errors were observed from a large code base, but I can try to make a minimal reproducer, if needed.

1. 
```
Tpetra_CrsMatrix_def.hpp:1133:41: error: operands to ?: have different types ‘Tpetra::CrsMatrix<double, long int, long long int>::local_matrix_type’ {aka ‘KokkosSparse::CrsMatrix<double, long int, Kokkos::OpenMP, void, long unsigned int>’} and ‘const local_matrix_type’ {aka ‘const KokkosSparse::CrsMatrix<double, int, Kokkos::OpenMP, void, long unsigned int>’}
     return lclMatrix_.get () == nullptr ?
Tpetra_CrsMatrix_def.hpp:1133:41: note:   and each type can be converted to the other
```
stemming from this piece of code:
https://github.com/trilinos/Trilinos/blob/9c0318f30ff284869ba160d851adaa45e09afe21/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp#L1035-L1043
where 
```
using local_matrix_type =
      KokkosSparse::CrsMatrix<matrix_scalar_type,
                              local_ordinal_type,
                              execution_space,
                              void,
                              typename local_graph_type::size_type>;
```
The issue is that in the `CrsMatrix`, `local_ordinal_type` is an alias for `LocalOrdinal` (the template parameter of the matrix), but in `LocalCrsMatrixOperator` (the type pointed to by `lclMatrix_`) it aliases `Tpetra::Details::DefaultTypes::local_ordinal_type` (which is hardcoded to `int`). 

The compiler hints at conversion, but my understanding is that would invoke a deep copy via `KokkosSparse::CrsMatrix`'s [conversion constructor](https://docs.trilinos.org/dev/packages/kokkos/doc/html/classKokkos_1_1CrsMatrix.html#ae4e491881a1b9207b0d305080be0cae8), were I to try and convert the second return value in the ternary. I'm not sure what the best way around it is. Perhaps I'm misunderstanding something?

I'm also getting some other related errors (e.g. in `Tpetra::CrsMatrix::fillLocalMatrix()`), but those are buried under a whole stack of `std::shared_ptr` stuff and are harder to parse. The fundamental problem is the same: `LocalCrsMatrixOperator` doesn't know about `LocalOrdinal` and uses a different type for the actual Kokkos object it wraps.

2. 
```
Tpetra_CrsMatrix_def.hpp:9650:69: error: operands to ?: have different types ‘Teuchos::ArrayView<const int>’ and ‘Teuchos::ArrayView<const long int>’
         Teuchos::ArrayView<const LO>   ELID1 = MyImporter.is_null() ? Teuchos::ArrayView<const int>() : MyImporter->getExportLIDs();
```
related to this:
https://github.com/trilinos/Trilinos/blob/9c0318f30ff284869ba160d851adaa45e09afe21/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp#L9605
Here a simple fix would be to replace `ArrayView<const int>` with `ArrayView<const LO>` in the rhs expression. It looks like that's what was intended there.

### Steps to Reproduce
1. SHA1:  55a75997332636a28afc9db1aee4ae46fe8d93e7 (aka `trilinos-release-12-18-1` tag)
1. Configure script: [trilinos_config.txt](https://github.com/trilinos/Trilinos/files/4951388/trilinos_config.txt)
1. Configure log: [trilinos_configure.log](https://github.com/trilinos/Trilinos/files/4951415/trilinos_configure.log)
1. Try to instantiate and use `Tpetra::CrsMatrix<double, std::ptrdiff_t, long long int>`

--- a/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp
@@ -9602,7 +9602,7 @@ namespace Tpetra {
         }
 
         Teuchos::ArrayView<const int>  EPID1 = MyImporter.is_null() ? Teuchos::ArrayView<const int>() : MyImporter->getExportPIDs();
-        Teuchos::ArrayView<const LO>   ELID1 = MyImporter.is_null() ? Teuchos::ArrayView<const int>() : MyImporter->getExportLIDs();
+        Teuchos::ArrayView<const LO>   ELID1 = MyImporter.is_null() ? Teuchos::ArrayView<const LO>() : MyImporter->getExportLIDs();
 
         Teuchos::ArrayView<const int>  TEPID2  =  rowTransfer.getExportPIDs(); // row matrix
         Teuchos::ArrayView<const LO>   TELID2  =  rowTransfer.getExportLIDs();

